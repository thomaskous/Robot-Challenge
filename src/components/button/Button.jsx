import './index.css'
import React from 'react'

/* A simple button component with a clickHandler */
function Button({ active = true, classes, clickHandler, label }) {
  return (
    <button
      className={`button ${classes || ''}`}
      onClick={clickHandler}
      disabled={!active}
    >
      {label}
    </button>
  );
}

export default Button;