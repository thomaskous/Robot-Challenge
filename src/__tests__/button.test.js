import React from 'react'
import { render } from '@testing-library/react'
import Button from '../components/button/Button'

describe('<Button />', () => {
  it('renders an active <Button />', () => {
    const { container, getByRole } = render(
      <Button
        label="Place"
        classes="input-group"
        clickHandler={() => { }} />
    );

    expect(getByRole('button').disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  })

  it('renders a disabled <Button />', () => {
    const { container, getByRole } = render(
      <Button
        label="Place"
        classes="input-group"
        active={false}
        clickHandler={() => { }} />
    );

    expect(getByRole('button').disabled).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  })
});