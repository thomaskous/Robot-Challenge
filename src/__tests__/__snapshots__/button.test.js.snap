// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a disabled <Button /> 1`] = `
<button
  class="button input-group"
  disabled=""
>
  Place
</button>
`;

exports[`<Button /> renders an active <Button /> 1`] = `
<button
  class="button input-group"
>
  Place
</button>
`;
